### DISCUSSION TITLE
What is the best way to filter channels by name

### DISCUSSION THREAD

**Discussion by lemberh** [GetStream/stream-chat-android#2085] [Q&A] [answered]
What would you guys recommend to use when I want to filter channels by name, I use direct channels the name is generated from the member's field.
I use android UI components for channels list,
in the example search searches for the messages in the channels 
I use for channels list https://getstream.io/chat/docs/sdk/android/ui/components/channel-list/

**Comment by leandroBorgesFerreira** [comment, marked as answer]
**Edited:**

Hello @lemberh. 

You can use:

```
Filters.autocomplete("name", "some name here")
```

To filter channels by name. Please let me know it that solves your problem. 

**Original:** 

Hello @lemberh. 

We currently do not support filtering by channel's name. Like: `Filters.contains("name", "Channel name")`. 

But, as you use direct channels, you could filter by the members of the channel. Like:

```
Filters.`in`("members", listOf("id1", "id2"))
```

You said: _in the example search searches for the messages in the channels_. What do you mean? I see an example that we are filtering by the `type` and we would like to have channels of the type `message`.

**Reply by lemberh** [reply]
I can search by member id but I don't know how to search by member name.
About searching for messages in the channel I meant this component from demo App 
https://github.com/GetStream/stream-chat-android/blob/main/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/channel/list/ChannelListFragment.kt
Here it is used `private val searchViewModel: SearchViewModel`
For now I have used `ChannelListView.ChannelListItemPredicate` for filtering the channels list, but it is not designed for search because every time the search query changes I have to create new object

**Reply by filbabic** [reply]
Hey @lemberh!

We recently implemented a search by member/user name in the channels on the Compose SDK:

```kotlin
            Filters.and(
                filter,
                Filters.or(
                    Filters.and(
                        Filters.autocomplete("member.user.name", searchQuery),
                        Filters.notExists("name")
                    ),
                    Filters.autocomplete("name", searchQuery)
                )
            )
```

This is what we use by default - we search by the channel name if it exists, or by member name if there is no channel name defined. Let me know if this works! :]

**Reply by tarunkhatri** [reply]
Hi @filbabic , I am not able to find above functionality in 
implementation 'io.getstream:stream-chat-android-ui-components:5.11.2'

this library, can you guys please implement search by username functionality in UI components library. It would be very helpful.
Thanks

**Reply by filbabic** [reply]
Hey Tarun!

You should be able to set the filters for the ChanneListViewModel to be matching the code above, by calling `setFilters`. it doesn't work out of the box for the search UI we provide, but it can be done by custom UI on your end.

As for supporting this in UI Components, we'll have to look into our plans for this quarter and if we can fit that feature.
